.github/workflows/ci.yml

```diff
@@ -29,6 +29,11 @@ jobs:
       - name: Run ruby tests
         run: bundle exec rake spec
 
+      - name: Run ruby tests (hard-mode with GC.stress)
+        run: bundle exec rake spec
+        env:
+          GC_STRESS: "true"
```

Comment:
Good idea! 👍 

Is there value in doing this on all platform and ruby version? Because this is slow, I was thinking we could do it only once on e.g. linux + ruby 3.1.

Reply:
It's not too slow, IMO. If it does get slow, we can change it to a limited matrix like you mentioned.